src/app/profile.js

```diff
@@ -0,0 +1,20 @@
+"use client"
+import {useUser} from "react";
+import RootLayout from './layout.js';
+ 
+const Profile = () => {
+  const { user } = useUser({ redirectTo: '/login' })
+
+  if (!user || user.isLoggedIn === false) {
+    return <RootLayout>Loading...</RootLayout>
+  }
+ 
+  return (
+    <RootLayout>
+      <h1>Your Profile</h1>

```

Comment:
i dont think this part is needed, all you where to do is to have a header component to have all your design part and render it in the layout page.